app/master/subjob.py

```diff
@@ -40,11 +41,15 @@ def _set_atom_state(self, state):
         for atom in self._atoms:
             atom.state = state
 
-    def mark_in_progress(self):
+    def mark_in_progress(self, slave):
```

Comment:
Do you think this is better vs just storing the slave_url? (Actual question -- I'm interested in your thinking. :smile:)